src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/Event/GetAzureRmRecoveryServicesAsrEvent.cs

```diff
@@ -0,0 +1,204 @@
+using Microsoft.Azure.Management.RecoveryServices.SiteRecovery.Models;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.SiteRecovery
+{
+    /// <summary>
+    ///     Retrieves Azure Site Recovery alert and nofification settings.
+    /// </summary>
+    [Cmdlet(
+        VerbsCommon.Get,
+        "AzureRmRecoveryServicesAsrEvent",
+        DefaultParameterSetName = ASRParameterSets.ByParam)]
+    [Alias("Get-ASREvent")]
+    [OutputType(typeof(IEnumerable<ASREvent>))]
+    public class GetAzureRmRecoveryServicesAsrEvents : SiteRecoveryCmdletBase
+    {
+        /// <summary>
+        ///     Gets or sets Resource Id.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.ByResourceId, Mandatory = true)]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        ///     Gets or sets Resource Id.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.ByFabricId, Mandatory = true)]
+        public string FabricId { get; set; }
+
+        /// <summary>
+        ///     Gets or sets Event name.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.ByName, Mandatory = true)]
+        public string Name { get; set; }
+
+        /// <summary>
+        ///     Gets or sets server name.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.ByParam)]
+        [Parameter(ParameterSetName = ASRParameterSets.ByFabricId)]
+        [ValidateNotNullOrEmpty]
+        public string AffectedObjectFriendlyName { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the Endtime.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.ByParam)]
+        [Parameter(ParameterSetName = ASRParameterSets.ByFabricId)]
+        [ValidateNotNullOrEmpty]
+        public DateTime EndTime { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the fabric.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.ByParam)]
+        [Parameter(ParameterSetName = ASRParameterSets.ByFabricId)]
+        [ValidateNotNullOrEmpty]
+        public ASRFabric Fabric { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the Severity.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.ByParam)]
+        [Parameter(ParameterSetName = ASRParameterSets.ByFabricId)]
+        [ValidateSet(
+            ASRHealthEventServerity.Critical,
+            ASRHealthEventServerity.Warning,
+            ASRHealthEventServerity.OK,
+            ASRHealthEventServerity.Unknown)]
+        public string Severity { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the Start time.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.ByParam)]
+        [Parameter(ParameterSetName = ASRParameterSets.ByFabricId)]
+        [ValidateNotNullOrEmpty]
+        public DateTime StartTime { get; set; }
```

Comment:
@viverm should there be a corresponding `EndTime` parameter?

Reply:
Already there . But reordered below StartTime to give better visibility.